x-pack/platform/plugins/shared/onechat/server/services/chat/utils/get_chat_model.ts

```diff
@@ -19,19 +19,14 @@ export const getChatModel$ = ({
   inference,
   span,
 }: {
-  connectorId?: string;
+  connectorId: string;
   request: KibanaRequest;
   inference: InferenceServerStart;
   span?: Span;
 }): Observable<InferenceChatModel> => {
   return defer(async () => {
-    let selectedConnectorId = connectorId;
-    if (!selectedConnectorId) {
-      const defaultConnector = await inference.getDefaultConnector(request);
-      selectedConnectorId = defaultConnector.connectorId;
-    }
-    span?.setAttribute('elastic.connector.id', selectedConnectorId);
-    return selectedConnectorId;
```

Comment:
With the connectorId no longer optional here, you can ditch the first part of the chain.

```ts
  return defer(async () => {
    span?.setAttribute('elastic.connector.id', selectedConnectorId);
    return  inference.getChatModel({
        request,
        connectorId: selectedConnectorId,
        chatModelOptions: {
          telemetryMetadata: MODEL_TELEMETRY_METADATA,
        },
      });
  });
```